frontend/src/components/crd/CustomResourceDetails.tsx

```diff
@@ -24,37 +23,45 @@ export default function CustomResourceDetails() {
   }>();
   const [crd, setCRD] = React.useState<CRD | null>(null);
   const [error, setError] = React.useState<ApiError | null>(null);
-
-  const { t } = useTranslation('glossary');
+  const [item, setItem] = React.useState<KubeCRD | null>(null);
+  const [detailsError, setDetailsError] = React.useState<ApiError | null>(null);
 
   const namespace = ns === '-' ? undefined : ns;
 
+  React.useEffect(() => {
+    if (!crd) return;
+
+    const versions: [string, string, string][] = (crd.jsonData as KubeCRD).spec.versions.map(
+      versionInfo => [crd.spec.group, versionInfo.name, crd.spec.names.plural]
+    );
+
+    const CRClass = makeCustomResourceClass(versions, !!namespace);
+    CRClass.useApiGet(setItem, crName, namespace, setDetailsError);
```

Comment:
I think this would constitue using a hook conditionally (not allowed in React), and the reason why I had two components.